### PR TITLE
UNITY-8057 Sage REST: Patch/Put for crossselling

### PR DESCRIPTION
**Pull request by bolsu**
UNITY-8057 Sage REST: Patch/Put for crossselling

### CODE REVIEW

**Review by mjger — APPROVED** [review]

File: _raml/apps/products/products.raml
Comment (dkluever22) [inline _raml/apps/products/products.raml:142]: To match the description of the two new requests, I would suggest to adapt the wording as follows:
```suggestion
    description: Returns information on the cross-selling assignments of a single product. *epagesNow
```
Because an ID is used in the request, I guess it refers to a single product instead of several ones.

File: _raml/apps/products/products.raml
Comment (dkluever22) [inline _raml/apps/products/products.raml:162]:
```suggestion
            example: !include examples/put_product_productid_crossselling_response.json
```
I'm not sure if I'm right but by comparing other `PUT` requests I think the example to refer to needs to be _!include examples/put_product_productid_crossselling_response.json_
**Note:** This example file doesn't exist yet and needs to be created first!

File: _raml/apps/products/products.raml
Comment (dkluever22) [inline _raml/apps/products/products.raml:163]:
```suggestion
            schema: !include schema/put_product_productid_crossselling.schema.json
```
I think the same thing applies to the schema file.
**Note:** This schema file doesn't exist yet and needs to be created first!

File: _raml/apps/products/products.raml
Comment (dkluever22) [inline _raml/apps/products/products.raml:166]: To match the wording of other `PATCH` requests I would suggest to change it as follows:
```suggestion
    description: Updates information on the cross-selling assignments for a single product with partial data according to [RFC 6902](https://tools.ietf.org/html/rfc6902). You can either add or remove certain attributes. *epagesNow
```
Furthermore, we need to add the information to which product this feature applies. Is it just *epagesNow, as the `GET` request, or also base?
However, I'm really unsure about the part "with partial data according to [RFC 6902](https://tools.ietf.org/html/rfc6902)." Is this applicable for this endpoint? I'm also wondering, if we need to add information about supported JSON-PATCH operations and attribute paths that allow updates via PATCH (see [example](https://developer.epages.com/apps/api-reference/patch-shopid-orders-orderid.html))

File: _raml/apps/products/products.raml
Comment (dkluever22) [inline _raml/apps/products/products.raml:176]:
```suggestion
            example: !include examples/patch_product_productid_crossselling_response.json
```
I'm not sure if I'm right but by comparing other `PATCH` requests I think the example to refer to needs to be _!include examples/patch_product_productid_crossselling_response.json_
**Note:** This example file doesn't exist yet and needs to be created first!

File: _raml/apps/products/products.raml
Comment (dkluever22) [inline _raml/apps/products/products.raml:177]:
```suggestion
            schema: !include schema/patch_product_productid_crossselling.schema.json
```
I think the same thing applies to the schema file.
**Note:** This schema file doesn't exist yet and needs to be created first!

**Review by dkluever22 — CHANGES_REQUESTED** [review]
@bolsu Hi Tim! I took a look at this PR and it might be that I'm totally wrong but I have noticed several things compared to other requests within the documentation and left you some comments. Please don't hesitate to reach out if you have questions.

File: _raml/apps/products/products.raml
Comment (bolsu) [reply _raml/apps/products/products.raml:152]: Updated!

It's funny, but the GET request actually works for Base as well, so this is more of a marketing thing than a technical thing ;)

File: _raml/apps/products/products.raml
Comment (bolsu) [reply _raml/apps/products/products.raml:142]: This PR does not change anything in relation to the GET request, but I've updated this anyway

File: _raml/apps/products/products.raml
Comment (bolsu) [reply _raml/apps/products/products.raml:162]: The reason I am reusing the get_* json file is because I return for PUT, the exact same thing as the GET request, guaranteed

File: _raml/apps/products/products.raml
Comment (bolsu) [reply _raml/apps/products/products.raml:163]: Please see above comment

File: _raml/apps/products/products.raml
Comment (bolsu) [reply _raml/apps/products/products.raml:166]: I'm fine with your wording, and the link to 'RFC 6902' doesn't hurt, although I have made a minor adjustment in 'add or remove certain attributes' to 'add or remove assignments'.
Because we're dealing with assignments, there is only one path and we already state that only add/remove are supported. Coupled with this API being a fairly niche one along with the example request, and I think it is ok with your update.

File: _raml/apps/products/products.raml
Comment (bolsu) [reply _raml/apps/products/products.raml:176]: Same reasoning as for PUT

File: _raml/apps/products/products.raml
Comment (bolsu) [reply _raml/apps/products/products.raml:177]: See above

**Comment by bolsu** [comment]
@dkluever22 Thanks for the review Daria! I've replied to your comments and pushed another commit.

**Comment by bolsu** [comment]
@dkluever22 The epages6 and epagesj PRs will soon be merged, if not already, so this PR can also be merged when ready, thank you!

File: _raml/apps/products/products.raml
Comment (dkluever22) [reply _raml/apps/products/products.raml:162]: Please see comment below

File: _raml/apps/products/products.raml
Comment (dkluever22) [reply _raml/apps/products/products.raml:163]: I've checked again with other endpoints and it seems that even though it is the same thing for the `GET` request, we still have two files (see screenshot). I know it's duplicated content with a different title but since we've done it like this before, I would like to keep it that way. I guess the same thing applies to the examples. 
Bildschirm­foto 2023-02-23 um 16 20 39

File: _raml/apps/products/products.raml
Comment (dkluever22) [reply _raml/apps/products/products.raml:176]: See comment for PATCH

File: _raml/apps/products/products.raml
Comment (dkluever22) [reply _raml/apps/products/products.raml:177]: See comment for PATCH

Comment (bolsu) [reply _raml/apps/products/products.raml]: Updated

File: _raml/apps/products/products.raml
Comment (bolsu) [reply _raml/apps/products/products.raml]: Updated

File: _raml/apps/products/products.raml
Comment (bolsu) [reply _raml/apps/products/products.raml]: Updated

File: _raml/apps/products/products.raml
Comment (bolsu) [reply _raml/apps/products/products.raml]: Updated

**Review by dkluever22 — APPROVED** [review]

**Comment by dkluever22** [comment]
@bolsu I'm afraid we will need to wait until the public rollout on March, 16th to merge this PR. This is the catch with the Now documentation.

**Comment by bolsu** [comment]
@dkluever22 No Problems, I'll leave it in your capable hands :)

**Pull request merged**
